### ISSUE TITLE
php7: trait and return type declaration

### ISSUE THREAD

**Issue by flexjoly** [phpDocumentor/phpDocumentor2#1784]
Hi,

It seems much of php7 is parsed.

Only 'trait' and the new '[return type declaration](http://php.net/manual/en/functions.returning-values.php#functions.returning-values.type-declaration)' is not supported.

Luckily the parsers just skips these files, and parses all others. 

But it would help very much if this could be fixed. 
Trait is the most important to me. I removed al return-type-declarations, since that is not crucial. But traits are to nice to remove ;)
Apigen already parses traits.... But i prefer phpDocumentor.

Thanks so far!
Greetz, flexjoly
a divaeloper ;-)

**Comment by Dragonrun1** [comment]
So any work on when the return type issue gets fixed. This has become a show stopper for me as all my development is for PHP 7.0+ and now I can't gen any docs for them :(

**Comment by jaapio** [comment]
No, we are not ready with V3 yet. Due a leak of time for all of us there is not very much progress at the moment. 

Can you provide some more information about what is going on?

**Comment by Dragonrun1** [comment]
Main thing I've noticed is if you use return hints on any method in a class the whole class breaks and after a few classes the whole thing breaks so basically I end up with nothing useful. Even just something that keeps it from breaking in the parse phase would be a big help.

**Comment by mvriel** [comment]
I haven't been able to reproduce this issue using the 2.9.0 version of phpDocumentor. Can you verify that you are running this version and if so, provide me with a short sample that exhibits the failure that you mention?

I have tried phpDocumentor on a codebase that is based on PHP 7 but that doesn't seem to give the issues that you mention.

**Comment by Dragonrun1** [comment]
You can try it with my GitHub project
https://github.com/Yapeal/yapeal-ng
That's at least one I know it was break on.

**Comment by ashnazg** [comment]
@mvriel I can generate a failure by running current master against the `lib/Configuration/LogWiring.php` file in that `yapeal-ng` repo, using PHP 7.0.15.  See if that helps you.

**Comment by ashnazg** [comment]
Note, the error I could generate here is the same `PhpParser\Node\Stmt\Class_ could not be converted to string` that's shown on #1777 ...

**Comment by LeonardChallis** [comment]
Using version 2.9.0

```
$ phpdoc --version
phpDocumentor version 2.9.0
```

Sample code:
```
<?php
class Test
{
    /**
     * Do something
     * 
     * Please do something with this method
     * 
     * @return float|null
     */
    public function doSomething(): ?float
    {
        //
    }
}
```

And the output:

```
$ phpdoc
Collecting files .. OK
Initializing parser .. OK
Parsing files
Parsing /some/path/lib/Shipping/Manager.php
Parse Error: Syntax error, unexpected ':', expecting ';' or '{' on line 29  No summary was found for this file
Storing cache in "/some/path/phpdoc/output/build" .. OK
```

Note, error doesn't appear on subsequent calls to `phpdoc`.

**Comment by jaapio** [comment]
Closing this issue since this is resolved in v3.0.0-alpha1 and will be part of v3.0.0 once released. 
Return types are not supported yet but keeping track of that in #1901
